ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/ServerHostUtils.kt

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.engine
+
+import kotlin.reflect.*
+import kotlin.reflect.jvm.*
+
+/**
+ * Obtain function FQName.
+ */
+internal fun Function<*>.methodName(): String {
+    val method = (this as? KFunction<*>)?.javaMethod ?: return "${javaClass.name}.invoke"
```

Comment:
What if a lambda is suspend? It should be invokeSuspend? Shoull we prohibit it or allow?

Reply:
It's a tricky question. We don't support it right now and it will not be breaking change if we support it in the future. I would keep it as it is and get more use cases before.